httpcache.go

```diff
@@ -186,70 +199,67 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 		}
 
 		resp, err = transport.RoundTrip(req)
-		if err == nil && req.Method == "GET" && resp.StatusCode == http.StatusNotModified {
+		if err != nil {
+			if cachedResp != nil && canStaleOnError(req, cachedResp.Header) {
```

Comment:
already in cond block when `cacheResp != nil` ? I didn't find any modifiers above this.

Reply:
> already in cond block when `cacheResp != nil`?

Good catch.  Fixed in 55ebf79 -> 9bccf17.